### PR TITLE
Wraps calls to symbol() and decimals() in try catch (Fixes #4107)

### LINKED ISSUE #4107: Sending non-compliant tokens crashes MetaMask
Currently crashes MetaMask in a non-recoverable way. ASAP needs fix.

Some tokens do not implement ERC20 the same way, like [EOS](https://etherscan.io/address/0x86fa049857e0209aa7d9e616f7eb3b3b78ecfdb0#code) and MKR.

These both use different data types for the transferred quantity.

Long term, we need a strategy to support varying token transfers, but in the short term, we need to fail gracefully and recover the UI.

### PR DESCRIPTION
**Pull request by danjm**
Fixes #4107

Includes a change to `send-v2.js` unrelated to the bug, but helpful for testing the bug. That change allows the user to send 0 tokens, which we want regardless.

sendtokenswithoutsymboldecimals

### CODE REVIEW

**Review by kumavis — CHANGES_REQUESTED** [review]
handle case where decimals is 0

**Review by kumavis — CHANGES_REQUESTED** [review]

File: ui/app/token-util.js [hunk 1]
```diff
@@ -1,33 +1,45 @@
...
-    ])
+    return tokens[address]
+  }
+}
 
-    const [ symbol = [], decimals = [] ] = result
+async function getSymbolAndDecimals (tokenAddress, existingTokens = []) {
+  const existingToken = existingTokens.find(({ address }) => tokenAddress === address)
+  if (existingToken) {
+    return {
```
Comment (whymarrh) [inline ui/app/token-util.js]: Could we return the `exisitingToken`?

File: ui/app/components/pending-tx/index.js [hunk 2]
```diff
@@ -89,19 +92,14 @@ PendingTx.prototype.setTokenData = async function () {
...
-    const results = await Promise.all([
-      token.symbol(),
-      token.decimals(),
-    ])
-    const [ symbol, decimals ] = results
-
-    if (symbol[0] && decimals[0]) {
+    const { symbol, decimals } = await getSymbolAndDecimals(txParams.to, existingTokens)
+
+    if (symbol && decimals) {
```
[the remark this answers is not in the document]
Comment (danjm) [reply ui/app/components/pending-tx/index.js]: Handled in b71dbf52

Note that change assumes that the contract `decimal` method will return a `uint` (which ethjs-abi will return to as a BN).

File: ui/app/token-util.js (hunk 1 above)
Comment (danjm) [reply ui/app/token-util.js]: Done in 9f9e8789

**Review by kumavis — APPROVED** [review]

**Pull request merged**